src/Passes/cmake/Library.cmake

```diff
@@ -0,0 +1,38 @@
+macro(list_source_files result  directory)
+  file(GLOB_RECURSE source RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${directory}/*.cpp)
+  set(filelist "")
+  foreach(child ${source})
+    if(NOT ${child} MATCHES "(/Tests/)")
+      list(APPEND filelist ${child})
+    endif()
+  endforeach()  
+
+  set(${result} ${filelist})
+endmacro()
+
+function (microsoft_add_library          
+          library)
+  list(REMOVE_AT ARGV 0)
```

Comment:
Why is the first element removed from `ARGV`? I don't see it used again anywhere, but might be missing it.

Reply:
At this point, this is mostly just book keeping on what was already consumed and not. The reason why it is there is that at a later point, we may choose to extend the command such that it becomes `microsoft_add_library(name dep1 dep2 ... depN)` and by removing the arguments which were already consumed the arg list is ready to do this. However, there is no decision to do one or the other thing atm., so I have just prepared it for it.